scrapy/core/downloader/handlers/s3.py

```diff
@@ -49,14 +47,32 @@ def __init__(self, settings, aws_access_key_id=None, aws_secret_access_key=None,
         # If no credentials could be found anywhere,
         # consider this an anonymous connection request by default;
         # unless 'anon' was set explicitly (True/False).
-        anon = kw.get('anon', None)
+        anon = kw.get('anon')
         if anon is None and not aws_access_key_id and not aws_secret_access_key:
             kw['anon'] = True
+        self.anon = kw.get('anon')
 
+        self._signer = None
         try:
-            self.conn = _S3Connection(aws_access_key_id, aws_secret_access_key, **kw)
-        except Exception as ex:
-            raise NotConfigured(str(ex))
+            import botocore.auth
+            import botocore.credentials
+        except ImportError:
+            if six.PY3:
+                raise NotConfigured("missing botocore library")
+            _S3Connection = get_s3_connection()
+            if _S3Connection is None:
+                raise NotConfigured("missing botocore or boto library")
+            try:
+                self.conn = _S3Connection(
+                    aws_access_key_id, aws_secret_access_key, **kw)
```

Comment:
so, this is only done for boto library, yes?

there is one detail here, because the keyword arguments allow you to set `is_secure=False`, proxy and debug options -- this won't be possible with botocore/boto3, right?